tests/queries/0_stateless/00727_concat.sql

```diff
@@ -1 +1,74 @@
-SELECT CONCAT('Hello', ', ', 'world!');
+-- Tags: no-fasttest
+-- no-fasttest: json type needs rapidjson library, geo types need s2 geometry
+
+-- not tested here: (Simple)AggregateFunction, Nested
+
+SET allow_experimental_object_type = 1;
+SET allow_suspicious_low_cardinality_types=1;
+
+SELECT '-- Const string + non-const arbitrary type';
+SELECT concat('With ', materialize(42 :: Int8));
+SELECT concat('With ', materialize(43 :: Int16));
+SELECT concat('With ', materialize(44 :: Int32));
+SELECT concat('With ', materialize(45 :: Int64));
+SELECT concat('With ', materialize(46 :: Int128));
+SELECT concat('With ', materialize(47 :: Int256));
+SELECT concat('With ', materialize(48 :: UInt8));
+SELECT concat('With ', materialize(49 :: UInt16));
+SELECT concat('With ', materialize(50 :: UInt32));
+SELECT concat('With ', materialize(51 :: UInt64));
+SELECT concat('With ', materialize(52 :: UInt128));
+SELECT concat('With ', materialize(53 :: UInt256));
+SELECT concat('With ', materialize(42.42 :: Float32));
+SELECT concat('With ', materialize(43.43 :: Float64));
+SELECT concat('With ', materialize(44.44 :: Decimal(2)));
+SELECT concat('With ', materialize(true :: Bool));
+SELECT concat('With ', materialize(false :: Bool));
+SELECT concat('With ', materialize('foo' :: String));
+SELECT concat('With ', materialize('bar' :: FixedString(3)));
+SELECT concat('With ', materialize('foo' :: Nullable(String)));
+SELECT concat('With ', materialize('bar' :: Nullable(FixedString(3))));
+SELECT concat('With ', materialize('foo' :: LowCardinality(String)));
+SELECT concat('With ', materialize('bar' :: LowCardinality(FixedString(3))));
+SELECT concat('With ', materialize('foo' :: LowCardinality(Nullable(String))));
+SELECT concat('With ', materialize('bar' :: LowCardinality(Nullable(FixedString(3)))));
+SELECT concat('With ', materialize(42 :: LowCardinality(Nullable(UInt32))));
+SELECT concat('With ', materialize('fae310ca-d52a-4923-9e9b-02bf67f4b009' :: UUID));
+SELECT concat('With ', materialize('2023-11-14' :: Date));
+SELECT concat('With ', materialize('2123-11-14' :: Date32));
+SELECT concat('With ', materialize('2023-11-14 05:50:12' :: DateTime));
+SELECT concat('With ', materialize('2023-11-14 05:50:12.123' :: DateTime64(3)));
```

Comment:
Better provide an explicit time zone to DateTime, otherwise timezone randomization in CI will make the test flaky: `DateTime64(3, 'Europe/Amsterdam')`